it/services/NotProcessedFilesServiceISpec.scala

```diff
@@ -0,0 +1,102 @@
+
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+
+import helpers.SDESStub
+import models.SDESNotificationRecord
+import models.notification.RecordStatusEnum
+import org.mongodb.scala.Document
+import org.scalatest.matchers.should.Matchers._
+import play.api.test.Helpers._
+import repositories.FileNotificationRepository
+import scheduler.ScheduleStatus
+import uk.gov.hmrc.mongo.lock.MongoLockRepository
+import utils.{IntegrationSpecCommonBase, LogCapturing}
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+class NotProcessedFilesServiceISpec extends IntegrationSpecCommonBase with LogCapturing {
+  val lockRepository = injector.instanceOf[MongoLockRepository]
+  val service = injector.instanceOf[NotProcessedFilesService]
+  val notificationRepo = injector.instanceOf[FileNotificationRepository]
+
+  class Setup {
+    await(notificationRepo.collection.deleteMany(Document()).toFuture())
+    await(lockRepository.collection.deleteMany(Document()).toFuture())
+    await(lockRepository.ensureIndexes)
+    await(lockRepository.collection.countDocuments().toFuture()) shouldBe 0
+  }
+
+  lazy val dateTimeOfNow: LocalDateTime = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS)
+
+  val notificationRecord: SDESNotificationRecord = SDESNotificationRecord(
+    reference = "ref",
+    status = RecordStatusEnum.PENDING,
+    numberOfAttempts = 1,
+    createdAt = LocalDateTime.of(2020,1,1,1,1),
+    updatedAt = LocalDateTime.of(2020,2,2,2,2),
+    nextAttemptAt = LocalDateTime.of(2020,3,3,3,3),
+    notification = sampleNotification
+  )
+
+  val pendingNotifications: Seq[SDESNotificationRecord] = Seq(
+    notificationRecord,
+    notificationRecord.copy(reference = "ref1", updatedAt = dateTimeOfNow, nextAttemptAt = LocalDateTime.of(2020,3,3,3,3)),
+    notificationRecord.copy(reference = "ref2", nextAttemptAt = dateTimeOfNow.minusMinutes(2), status = RecordStatusEnum.FILE_RECEIVED_IN_SDES),
+    notificationRecord.copy(reference = "ref3", nextAttemptAt = LocalDateTime.of(2020,3,3,3,3), status = RecordStatusEnum.FILE_RECEIVED_IN_SDES),
+    notificationRecord.copy(reference = "ref4", nextAttemptAt = LocalDateTime.of(2020,3,3,3,3), status = RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY)
+  )
+
+  "tryLock" should {
+    "not do anything if the job is already locked" in new Setup {
+      val randomServerId = "123"
+      val releaseDuration: FiniteDuration = 123.seconds
+      await(lockRepository.collection.countDocuments().toFuture()) shouldBe 0
+      await(lockRepository.takeLock(service.lockKeeper.lockId, randomServerId, releaseDuration))
+      await(lockRepository.collection.countDocuments().toFuture()) shouldBe 1
+
+      await(service.invoke).getOrElse("fail") shouldBe s"${service.jobName} - JobAlreadyRunning"
+      await(lockRepository.collection.countDocuments().toFuture()) shouldBe 1
+    }
+  }
+
+  "invoke" should {
+    "run the job successfully if there is not notifications" in new Setup {
```

Comment:
Sounds like there is a dependency on there being notifications:
```suggestion
    "run the job successfully even if there is no notification" in new Setup {
```